ethdb/memorydb/memorydb.go

```diff
@@ -125,12 +124,12 @@ func (db *Database) Delete(key []byte) error {
 // DeleteRange deletes all of the keys (and values) in the range [start,end)
 // (inclusive on start, exclusive on end).
 func (db *Database) DeleteRange(start, end []byte) error {
-	it := db.NewIterator(nil, start)
-	defer it.Release()
+	db.lock.Lock()
+	defer db.lock.Unlock()
 
-	for it.Next() && bytes.Compare(end, it.Key()) > 0 {
```

Comment:
I think this method was bugged previously:  `bytes.Compare(end, it.Key()) > 0` will be true on the `end` key, leading to it being deleted.  Whereas the method description says that it should exclude end.

Reply:
Ah no.  I'm wrong.